connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -472,14 +501,48 @@ public String toString() {
         return newConsumer;
     }
 
+    protected <V> Result<V> execute(Operation<V> operation) {
```

Comment:
Can this method be moved to super class? Also would it be better to pass the success and failure operations as call back as well to the executor?

Reply:
in most cases, we simply need to skip records. not sure if that logic can be passed into the executor. it's the sink/source task that needs to execute that piece.